linker/src/lib.rs

```diff
@@ -93,31 +93,32 @@ pub fn link<T: FieldElement>(graph: PILGraph<T>) -> Result<PILFile<T>, Vec<Strin
             let to = link.to;
 
             // the lhs is `instr_flag { inputs, outputs }`
+            let mut expressions =
+                Vec::with_capacity(1 + from.params.inputs.len() + from.params.outputs.len());
```

Comment:
I don't think it's worth it reserving the capacity.

Reply:
I can remove it, but it doesn't hurt either... I've added it because I broke the single iterator chaining call into a few extends to make it more readable.